File: app/src/main/java/example/dgorod/widget/data/api/ApiInterface.java
package example.dgorod.widget.data.api;

import java.util.List;

import example.dgorod.widget.data.api.dto.GistDto;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * @author Dmytro Gorodnytskyi
 *         on 18-Apr-17.
 */

public interface ApiInterface {

    @GET("users/{username}/gists")
    Single<List<GistDto>> getGists(@Path("username") String username);
}

File: app/src/main/java/example/dgorod/widget/data/api/ApiInterface.kt
package example.dgorod.widget.data.api

import example.dgorod.widget.data.api.dto.GistDto
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

/**
 *
 * @author Dmytro Gorodnytskyi
 * on 23-May-17.
 */
interface ApiInterface {

    @GET("users/{username}/gists")
    fun getGists(@Path("username") username: String): Single<List<GistDto>>
}